agent/api/task/task.go

```diff
@@ -1753,7 +1795,7 @@ func (task *Task) dockerExposedPorts(container *apicontainer.Container) (dockerE
 			}
 			// expose egress listener port if present
 			ec := task.ServiceConnectConfig.EgressConfig
-			if ec.ListenerName != "" { // it's possible that task does not have an egress listener
+			if ec != nil && ec.ListenerName != "" { // it's possible that task does not have an egress listener
```

Comment:
I think we can get rid of `ec.ListenerName != ""` :) 
Back then egress config can never be nil so this is the only way checking whether one has been actually specified.

Reply:
Updated the if statement in the commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on agent/api/task/task.go. Thanks.